entrypoint.sh

```diff
@@ -407,12 +413,38 @@ runPostSetupScripts() {
     set -e
     echo "Post setup scripts execution succeeded."
 }
+function runCertbotAsNeeded() {
+    if [ ! "$GENERATE_CERTBOT_CERT_SCHEDULED" = "True" ]; then
+        echo "No scheduled cert creation."
+        return
+    fi
+
+    echo "Waiting for Zulip to come online to generate cert ..."
+    while ! curl -sk "$SETTING_EXTERNAL_HOST" >/dev/null 2>&1; do
+        sleep 1;
+    done
+
+    echo "Generating LetsEncrypt cert ..."
+
+    # Remove the self-signed certs which were only needed to get Zulip going.
+    rm -f "$DATA_DIR"/certs/zulip.key "$DATA_DIR"/certs/zulip.combined-chain.crt
+
+    # Accept the terms of service automatically.
+    /home/zulip/deployments/current/scripts/setup/setup-certbot \
+        --agree-tos \
+        --hostname="$SETTING_EXTERNAL_HOST" \
+        --email="$SETTING_ZULIP_ADMINISTRATOR" \
+        --deploy-hook "$ZULIP_CERTBOT_DEPLOY_HOOK"
+
+    echo "LetsEncrypt cert generated."
+}
 bootstrappingEnvironment() {
     echo "=== Begin Bootstrap Phase ==="
     waitingForDatabase
     zulipFirstStartInit
     zulipMigration
     runPostSetupScripts
+    runCertbotAsNeeded &
```

Comment:
I'm OK with merging this approach for v1 since it is way better than not having the feature, but I think for v2, it might be better to e.g. run with `certbot --standalone` for the initial certificate creation, before the container tries to bring `nginx` up, rather than this run-in-background approach (that would also let us get rid of the self-signed certificate piece, I think?).  i.e. before the `exec supervisord` line that boots `nginx`.  Thoughts?